chainer/functions/connection/convolution_nd.py

```diff
@@ -0,0 +1,354 @@
+                    handle, self.filter_desc.value, gy_desc.value,
+                    self.conv_desc.value, x_desc.value, _bwd_data_pref,
+                    workspace_size)
+                libcudnn.convolutionBackwardData_v3(
+                    handle, one.data, self.filter_desc.value, W.data.ptr,
+                    gy_desc.value, gy.data.ptr, self.conv_desc.value,
+                    algo, workspace.data.ptr, workspace_size,
+                    zero.data, x_desc.value, gx.data.ptr)
+            else:
+                libcudnn.convolutionBackwardFilter_v2(
+                    handle, one.data, x_desc.value, x.data.ptr,
+                    gy_desc.value, gy.data.ptr, self.conv_desc.value,
+                    zero.data, self.filter_desc.value, gW.data.ptr)
+                libcudnn.convolutionBackwardData_v2(
+                    handle, one.data, self.filter_desc.value, W.data.ptr,
+                    gy_desc.value, gy.data.ptr, self.conv_desc.value,
+                    zero.data, x_desc.value, gx.data.ptr)
+
+            if b is not None:
+                gb = cuda.cupy.empty_like(b)
+                libcudnn.convolutionBackwardBias(
+                    handle, one.data, gy_desc.value, gy.data.ptr,
+                    zero.data, self.bias_desc.value, gb.data.ptr)
+        # Implementation using col2im.
+        else:
+            gW_mat = gW.reshape(out_c, reduce(operator.mul, ksize, c))
+            col_mats = self.col.reshape(
+                n, reduce(operator.mul, ksize, c), reduce(operator.mul, outs))
+            gy_mats = gy.reshape(n, out_c, reduce(operator.mul, outs))
+            # TODO(takagi): Use streams or batch gemm
+            gW_mat[...] = 0
+            for i in moves.range(n):
+                gW_mat += cuda.cupy.dot(gy_mats[i], col_mats[i].T)
+
+            # Compute patch array gradient.
+            W_mat = W.reshape(out_c, -1)
+            gcol = cuda.cupy.empty_like(self.col)
+            gcol_mats = gcol.reshape(
+                n, reduce(operator.mul, ksize, c), reduce(operator.mul, outs))
+            for i in moves.range(n):
+                gcol_mats[i] = cuda.cupy.dot(W_mat.T, gy_mats[i])
+
+            # Compute input gradient.
+            gx = conv_nd.col2im_nd_gpu(gcol, stride, pad, dims)
+
+            # Compute bias gradient if given.
+            if b is not None:
+                # (n, _, out_1, out_2, ..., out_N)
+                axis = (0,) + tuple(moves.range(2, ndim + 2))
+                gb = gy.sum(axis=axis)
+
+        if b is None:
+            return gx, gW
+        else:
+            return gx, gW, gb
+
+
+def convolution_nd(x, W, b=None, stride=1, pad=0, use_cudnn=True,
+                   cover_all=False):
+    """N-dimensional convolution function.
+
+    This is an implementation of N-dimensional convolution which is generalized
+    two-dimensional convolution in ConvNets. It takes three variables: the
+    input ``x``, the filter weight ``W`` and the bias vector ``b``.
```

Comment:
I've generalized it with just saying "the input ``x``" in the fix commit, because I do not have good idea: volume, tensor, data, body, .... Do you have one?